sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -1112,19 +1109,20 @@ private void newTimeout(
         this.acquisitionAndIdleEndpointDetectionTimeout.set(acquisitionAndIdleEndpointDetectionTimer.newTimeout(
             (Timeout timeout) -> {
                 if (idleEndpointTimeoutInNanos == 0) {
-                    if (logger.isDebugEnabled()) {
-                        logger.debug("Idle endpoint check is disabled");
+                    // log is too noisy when debug is enabled
+                    if (logger.isTraceEnabled()) {
```

Comment:
NIT (not necessary to fix) - is xxxEnabled is only necessary if one of the parameter values is non-trivial to calculate. Below the Duration calculation for example makes sense to exclude. The actual string formatting is gated behind level check anyway. So, for this trace the explicit check is not strictly necessary - but I don't mind either. So fixing it is optional.